app/javascript/hooks/use-history.ts

```diff
@@ -11,28 +11,36 @@ function removeEmpty<TParams>(obj: TParams) {
         return v !== null
       }
     })
-    .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {})
+    .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {}) as TParams
 }
 
 function toQuery<TParams>(params: TParams): string {
-  const state = removeEmpty(params)
-
-  return `?${stringify(decamelizeKeys(state), { arrayFormat: 'brackets' })}`
+  return `?${stringify(decamelizeKeys(params as Record<string, unknown>), {
+    arrayFormat: 'brackets',
+  })}`
 }
 
 function pushState<TParams>(params: TParams) {
   history.pushState(history.state, '', toQuery(params))
 }
 
 export function useHistory<TParams>({ pushOn }: { pushOn: TParams }): void {
+  const [history, setHistory] = useState(removeEmpty(pushOn))
```

Comment:
I don't quite understand what this is supposed to do. Can you tell me why there's this extra internal state?

Reply:
Thanks. I've rewritten the code and removed it.